CRM/Report/Form.php

```diff
@@ -4893,4 +4893,26 @@ public function joinGroupTempTable($baseTable, $field, $tableAlias) {
     }
   }
 
+  /**
+   * Get all labels for fields that are used in a group concat.
+   *
+   * @param string $options
+   *   comma separated option values.
+   * @param array $group
+   *   array of option value/label pairs.
+   *
+   * return string
+   */
+  public function getGroupLabels($options, $group) {
```

Comment:
Oh I see, my comments were more about the signature of your function than the workings inside it. But, to strip duplicates one trick I like is to add them to the keys eg.

$labels[$label] =TRUE;

& then at the end do return array_keys($label).

My main thinking thought was to resolve the options within the function rather than passing them in as an array